content/docs/rfcs/59/README.md

```diff
@@ -0,0 +1,153 @@
+---
+slug: 59
+title: 59/STATUS-URL-DATA
+name: Status URL Data
+status: raw
+category: Standards Track
+tags: waku-application
+editor: Felicio Mununga <felicio@status.im>
+contributors:
+  - Aaryamann Challani <aaryamann@status.im>
+---
+
+# Abstract
+
+This document specifies serialization, compression, and encoding techniques used to transmit data within URLs in the context of Status protocols.
+
+# Motivation
+
+When sharing URLs, link previews often expose metadata to the websites behind those links.
+To reduce reliance on external servers for providing appropriate link previews, this specification proposes a standard method for encoding data within URLs.
+
+# Terminology
+
+- Community: Refer to [56/STATUS-COMMUNITIES](/spec/56)
+- Channel: Refer to terminology in [56/STATUS-COMMUNITIES](/spec/56)
+- User: Refer to terminology in [56/STATUS-COMMUNITIES](/spec/56)
+- Shard Refer to terminology in [51/WAKU2-RELAY-SHARDING](/spec/51)
+
+# Wire Format
+
+```protobuf
+syntax = "proto3";
+
+message Community {
+    // Display name of the community
+    string display_name = 1;
+    // Description of the community
+    string description = 2;
+    // Number of members in the community
+    uint32 members_count = 3;
+    // Color of the community title
+    string color = 4;
+    // List of tag indices
+    repeated uint32 tag_indices = 5;
+}
+
+message Channel {
+    // Display name of the channel
+    string display_name = 1;
+    // Description of the channel
+    string description = 2;
+    // Emoji of the channel
+    string emoji = 3;
+    // Color of the channel title
+    string color = 4;
+    // Community the channel belongs to
+    Community community = 5;
+    // UUID of the channel
+    string uuid = 6;
+}
+
+message User {
+    // Display name of the user
+    string display_name = 1;
+    // Description of the user
+    string description = 2;
+    // Color of the user title
+    string color = 3;
+}
+
+message URLData {
+    // Community, Channel, or User
+    bytes content = 1;
+    uint32 shard_cluster = 2;
+    uint32 shard_index = 3;
+}
```

Comment:
@richard-ramos @cammellos updated the spec in https://github.com/vacp2p/rfc/pull/600/commits/1c38bb11369ce03e30cb50e53077ec5d3f0b7da1.

I forgot to ask, do you think that these fields (and their "informational value") should be after "#" (i.e. not exposed to servers) as well as the public keys are now, or don't have to be?

Reply:
IMO with the idea of reducing as much metadata as possible stored in the logs of servers, these fields should appear after the #  but I don't know the ramifications this could cause?. Do the URLs for sharing communities with its public key follow this format?

Reply:
> Do the URLs for sharing communities with its public key follow this format?

If you mean if they are after the #, then yes. Merged implementations up until now should follow https://github.com/vacp2p/rfc/pull/602/files.

> I don't know the ramifications this could cause?

When I asked you, I could only think of targeting the community's topic. Given that public key isn't needed to construct it. @rymnc @cammellos @jakubgs could you weigh in as well please?

Latest privacy consideration on this topic being mentioned in https://github.com/status-im/status-web/issues/327#issuecomment-1470453154.

> these fields should appear after the #

Given above, I'd agree. In which case I'd propose mapping it with the public key and a new protobuf, similarly to how the content is encoded.

Reply:
>> these fields should appear after the #
>
>Given above, I'd agree. In which case I'd propose mapping it with the public key and a new protobuf, similarly to how the content is encoded.

@cammellos agrees.